### DISCUSSION TITLE
Object X and Objext Y in the FROM clause have the same exposed names. Use correlation names to distinguish them.

### DISCUSSION THREAD

**Discussion by maticjosip** [sqlalchemy/sqlalchemy#9315] [Usage Questions] [answered]
I have two objects with the same table name but a different schema name (Stage/Destination). Why does this error occurs when the displayed SQL Query is correct even including the different schema names to the join.

sqlalchemy.exc.ProgrammingError: (pyodbc.ProgrammingError) ('42000', '[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]The objects "stg.table" and "dest.table" in the FROM clause have the same exposed names. Use correlation names to distinguish them. (1013) (SQLExecDirectW)')
[SQL: DELETE FROM [dest].[table] OUTPUT deleted.id FROM [dest].[table], [stg].[table] WHERE [dest].[table].[colA] = [stg].[table].[colA] AND [dest].[table].[colB] = [stg].[table].[colB]]
(Background on this error at: https://sqlalche.me/e/20/f405)

**Comment by zzzeek** [comment, marked as answer]
the two tables have the same name and SQL Server is not smart enough to distinguish them on the schema name, so the table that's not the subject of the delete you would want to apply aliased() to it.

**Reply by maticjosip** [reply]
Thanks!
